playground/backend/internal/preparators/java_preparators.go

```diff
@@ -0,0 +1,125 @@
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preparators
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	classWithPublicModifierPattern    = "public class "
+	classWithoutPublicModifierPattern = "class "
+	packagePattern                    = `package ([\w]+\.)+[\w]+;`
+	emptyStringPattern                = ""
+	newLinePattern                    = "\n"
+	pathSeparatorPattern              = "/"
+	tmpFileSuffix                     = "tmp"
+)
+
+// GetJavaPreparation return preparation methods that should be applied to Java code
+func GetJavaPreparation(filePath string) *[]Preparator {
+	publicClassModification := Preparator{
+		Prepare: replace,
+		Args:    []interface{}{filePath, classWithPublicModifierPattern, classWithoutPublicModifierPattern},
+	}
+	additionalPackage := Preparator{
+		Prepare: replace,
+		Args:    []interface{}{filePath, packagePattern, emptyStringPattern},
+	}
+	preparation := []Preparator{publicClassModification, additionalPackage}
+	return &preparation
+}
+
+// replace process file by filePath and replace all patterns to newPattern
+func replace(args ...interface{}) error {
+	filePath := args[0].(string)
+	pattern := args[1].(string)
+	newPattern := args[2].(string)
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		log.Printf("Preparation: Error during open file: %s, err: %s\n", filePath, err.Error())
+		return err
+	}
+	defer file.Close()
+
+	tmp, err := createTempFile(filePath)
+	if err != nil {
+		log.Printf("Preparation: Error during create new temporary file, err: %s\n", err.Error())
+		return err
+	}
+	defer tmp.Close()
+
+	reg := regexp.MustCompile(pattern)
+	scanner := bufio.NewScanner(file)
+
+	// uses to indicate when need to add new line to tmp file
+	newLine := false
+	for scanner.Scan() {
+		err = addNewLine(newLine, tmp)
+		if err != nil {
+			log.Printf("Preparation: Error during write \"%s\" to tmp file, err: %s\n", newLinePattern, err.Error())
+			return err
+		}
+		line := scanner.Text()
+		matches := reg.FindAllString(line, -1)
+		for _, str := range matches {
+			line = strings.ReplaceAll(line, str, newPattern)
+		}
+		if _, err = io.WriteString(tmp, line); err != nil {
+			log.Printf("Preparation: Error during write \"%s\" to tmp file, err: %s\n", line, err.Error())
+			return err
+		}
+		newLine = true
+	}
```

Comment:
Let's make this block a separate function – seems like we can easily extract it and simplify the `replace` function.